www/index.js

```diff
@@ -124,6 +120,28 @@ monaco.languages.onLanguage(modeId, async () => {
             }
         },
     });
+    monaco.languages.registerInlayHintsProvider(modeId, {
+        async provideInlayHints(model, range, token) {
+            let hints = await state.inlay_hints();
+            return hints.map((hint) => {
+                if (hint.hint_type == 1) {
```

Comment:
What about hint type 0?

Reply:
RA doesn't emit type 0 (type 0 is other in lsp/monaco):

```rust
                match ih.kind {
                    InlayKind::TypeHint | InlayKind::ChainingHint => InlayHintType::Type,
                    InlayKind::ParameterHint => InlayHintType::Parameter,
                },
```

Reply:
Could you remove that enum variant on the rust side? You can explicitly specify a discriminant to keep the numbers identical.

Reply:
I didn't get exactly what you want. `InlayHintType` is not meaningless and is the rust equivalent of this enum: https://microsoft.github.io/monaco-editor/api/enums/monaco.languages.inlayhintkind.html

You want to send `ih.kind` and move this `match` to JS?

Reply:
If `InlayHintType::Other` is not handled here I think it is best to remove this variant completely and only re-introduce it once it is being used. I think that will save some debugging when it becomes used.

Reply:
Aha, just the variant. I thought the whole enum should be removed.